src/cloud-element-templates/util/propertyUtil.js

```diff
@@ -353,38 +353,39 @@ export function setPropertyValue(bpmnFactory, commandStack, element, property, v
   // property
   if (PROPERTY_BINDING_TYPES.includes(type)) {
 
-    const propertyDescriptor = businessObject.$descriptor.propertiesByName[ name ];
+    const propertyType = businessObject.$descriptor.propertiesByName[ name ]?.type;
+    let updatedProperty = name;
 
-    // if property not created yet
-    if (!propertyDescriptor) {
+    if (!propertyType || propertyType === 'String') {
 
-      // make sure we create the property
+      // make sure we create and don't remove the property
       propertyValue = value || '';
-    }
+    } else if (propertyType === 'Boolean') {
+      propertyValue = !!value;
+    } else if (propertyType === 'Integer') {
+      propertyValue = parseInt(value, 10);
 
-    else {
-      const { type: propertyType } = propertyDescriptor;
+      if (isNaN(propertyValue)) {
 
-      // do not override non-primitive types
-      if (!PRIMITIVE_MODDLE_TYPES.includes(propertyType)) {
-        throw new Error(`cannot set property of type <${ propertyType }>`);
+        // do not set NaN value
+        propertyValue = undefined;
       }
+    } else if (EXPRESSION_TYPES.includes(propertyType)) {
+      const existingExpression = businessObject.get(name);
 
-      if (propertyType === 'Boolean') {
-        propertyValue = !!value;
-      } else if (propertyType === 'Integer') {
-        propertyValue = parseInt(value, 10);
-
-        if (isNaN(propertyValue)) {
-
-          // do not set NaN value
-          propertyValue = undefined;
-        }
-      } else {
+      if (existingExpression && is(existingExpression, 'bpmn:FormalExpression')) {
 
-        // make sure we don't remove the property
+        // re-use existing expression
+        businessObject = existingExpression;
+        updatedProperty = 'body';
         propertyValue = value || '';
+      } else {
+        propertyValue = createExpression(value, businessObject, bpmnFactory);
       }
+    } else {
+
+      // unsupported non-primitive types cannot be set
+      throw new Error(`cannot set property of type <${ propertyType }>`);
```

Comment:
Good. We could test cover this.

Reply:
Done via https://github.com/bpmn-io/bpmn-js-element-templates/pull/161/commits/9c5619ea5fe3cd55b5aef9a17af619b916eaadae